### DISCUSSION TITLE
How to get the modularity of each community

### DISCUSSION THREAD

**Discussion by chengaofeng1** [networkit/networkit#692] [Q&A]
by using  community.Modularity().getQuality(communities, G),   we can only get the sum modularity of all community,  how to get the modularity of each community?

**Comment by chengaofeng1** [comment]
anyone who can help me? thanks a lot

**Comment by avdgrinten** [comment]
How exactly do you define the modularity of a single cluster? The contribution to the modularity by all nodes of that cluster? Adding this to the NetworKit API would be simple. However, it's not immediately clear to me that this is a useful measure.

**Comment by chengaofeng1** [comment]
yes, the modularity of one community is the contribution to the sum modularity of all community by all nodes of this community. This measure can be used as a indicator of the importance of each community. 
So I hope that you will complete this indicator as soon as possible. Thanks.

**Comment by hmeyerhenke** [comment]
Due to other urgent tasks, we cannot implement this request in the next days. However, NetworKit is open source and the algorithmic change is not very big. You might want to try yourself to speed things up... Look into the community module and there into the Modularity class.

**Comment by chengaofeng1** [comment]
so embarrassed now，because I am not good at C++. When you have time to implement this request , please let me know. Thanks a lot.
